annotation-file-utilities/src/annotator/find/AnnotationInsertion.java

```diff
@@ -102,13 +107,45 @@ protected String getText(boolean comments, boolean abbreviate) {
     }
 
     /**
+     * Returns the fully-qualified name of the annotation, given its string representation.
+     * For example, given "@com.foo.Bar(baz)", returns "com.foo.Bar".
+     *
+     * @param annotation the string representation of the annotation; starts with "@"
+     * @return the fully-qualified name of the annotation, given its string representation
+     */
+    private static String extractAnnotationFullyQualifiedName(String annotation) {
```

Comment:
In new PR #136 I've added an assertion as pre-condition check on the passed argument `annotation` must be a fully qualified name in the constructor of `AnnotationInsertion`.

This change will restricts clients of `AnnotationInsertion` must passing a fully-qualified annotation name when creating an `AnnotationInsertion`.

Therefore, I can guarantee `AnnotationInsertion#extractAnnotationFullyQualifiedName` would extract the fully-qualified name of the annotation, by making this method non-static and extract the name from the final field `fullyQualifiedAnnotation`. (see my new PR for this change).